src/Http/ExpoController.php

```diff
@@ -16,72 +17,31 @@ public function __construct(
     ) {
     }
 
-    public function subscribe(Request $request): JsonResponse
+    public function subscribe(SubscribeRequest $request): ExpoSubscribeResource
     {
-        $validator = Validator::make($request->all(), [
-            'expo_token'    =>  'required|string',
-        ]);
-
-        if ($validator->fails()) {
-            return new JsonResponse([
-                'status' => 'failed',
-                'error' => [
-                    'message' => 'Expo Token is required',
-                ],
-            ], 422);
-        }
-
-        $token = $request->get('expo_token');
-
-        $interest = $this->expoChannel->interestName(Auth::user());
-
-        try {
-            $this->expoChannel->expo->subscribe($interest, $token);
-        } catch (\Exception $e) {
-            return new JsonResponse([
-                'status'    => 'failed',
-                'error'     =>  [
-                    'message' => $e->getMessage(),
-                ],
-            ], 500);
-        }
-
-        return new JsonResponse([
-            'status'    =>  'succeeded',
-            'expo_token' => $token,
-        ], 200);
+        $this->expoChannel
+            ->expo
+            ->subscribe($this
+                ->expoChannel
+                ->interestName(Auth::user()), $request->validated('expo_token')
+            );
+
+        return ExpoSubscribeResource::make($request->onlyValidated());
     }
 
-    public function unsubscribe(Request $request): JsonResponse
+    public function unsubscribe(UnsubscribeRequest $request): ExpoUnsubscribeResource
     {
-        $interest = $this->expoChannel->interestName(Auth::user());
-
-        $validator = Validator::make($request->all(), [
-            'expo_token'    =>  'sometimes|string',
-        ]);
-
-        if ($validator->fails()) {
-            return new JsonResponse([
-                'status' => 'failed',
-                'error' => [
-                    'message' => 'Expo Token is invalid',
-                ],
-            ], 422);
-        }
+        $interest = $this
```

Comment:
We have auth user in request, please use ready auth user object from $request.
```suggestion
        $interest = $this->expoChannel->interestName($request->user());
```